feature/experimental/gribi/ate_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,655 @@
+		vlanID := uint16(i) + 1
+		name := fmt.Sprintf(`dst%d`, i)
+		Index := uint32(i)
+		ateIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 1))
+		dutIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 2))
+		configureSubinterfaceDUT(t, d, dutPort, Index, vlanID, dutIPv4)
+		configureATE(t, top, atePort, name, vlanID, dutIPv4, ateIPv4+"/30")
+		nextHops = append(nextHops, ateIPv4)
+	}
+	configureInterfaceDUT(t, dutPort, d, "dst")
+	pushConfig(t, dut, dutPort, d)
+	return nextHops
+}
+
+// configureSubinterfaceDUT configures a single DUT layer 3 sub-interface.
+func configureSubinterfaceDUT(t *testing.T, d *oc.Root, dutPort *ondatra.Port, index uint32, vlanID uint16, dutIPv4 string) {
+	t.Helper()
+
+	i := d.GetOrCreateInterface(dutPort.Name())
+	s := i.GetOrCreateSubinterface(index)
+	if vlanID != 0 {
+		if *deviations.DeprecatedVlanID {
+			s.GetOrCreateVlan().VlanId = oc.UnionUint16(vlanID)
+		} else {
+			s.GetOrCreateVlan().GetOrCreateMatch().GetOrCreateSingleTagged().VlanId = ygot.Uint16(vlanID)
+		}
+	}
+
+	sipv4 := s.GetOrCreateIpv4()
+
+	if *deviations.InterfaceEnabled && !*deviations.IPv4MissingEnabled {
+		sipv4.Enabled = ygot.Bool(true)
+	}
+	a := sipv4.GetOrCreateAddress(dutIPv4)
+	a.PrefixLength = ygot.Uint8(uint8(ipv4PrefixLen))
+}
+
+// configureATE configures a single ATE layer 3 interface.
+func configureATE(t *testing.T, top *ondatra.ATETopology, atePort *ondatra.Port, Name string, vlanID uint16, dutIPv4, ateIPv4 string) {
+	t.Helper()
+
+	i := top.AddInterface(Name).WithPort(atePort)
+	if vlanID != 0 {
+		i.Ethernet().WithVLANID(vlanID)
+	}
+	i.IPv4().WithAddress(ateIPv4)
+	i.IPv4().WithDefaultGateway(dutIPv4)
+}
+
+// awaitTimeout calls a fluent client Await, adding a timeout to the context.
+func awaitTimeout(ctx context.Context, c *fluent.GRIBIClient, t testing.TB, timeout time.Duration) error {
+	subctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	return c.Await(subctx, t)
+}
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	ctx        context.Context
+	client     *fluent.GRIBIClient
+	dut        *ondatra.DUTDevice
+	ate        *ondatra.ATEDevice
+	top        *ondatra.ATETopology
+	electionID gribi.Uint128
+}
+
+// createTrafficFlow generates traffic flow from source network to destination network via
+// srcEndPoint to dstEndPoint and checks for packet loss.
+func createTrafficFlow(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology, flowArgs *flowArgs) *ondatra.Flow {
+	ethHeader := ondatra.NewEthernetHeader()
+	ipv4Header := ondatra.NewIPv4Header()
+	ipv4Header.DstAddressRange().
+		WithMin(flowArgs.flowStartAddress).
+		WithMax(flowArgs.flowEndAddress).
+		WithCount(flowArgs.flowCount)
+
+	srcEndPoint := top.Interfaces()["src"]
+	dstEndPoint := []ondatra.Endpoint{}
+	for intf, intfData := range top.Interfaces() {
+		if intf != "src" {
+			dstEndPoint = append(dstEndPoint, intfData)
+		}
+	}
+
+	flow := ate.Traffic().NewFlow("Flow").
```

Comment:
might as well collapse into one line:
return ate.Traffic() ...

Reply:
updated, thank you